src/diffpy/utils/diffraction_objects.py

```diff
@@ -186,11 +194,16 @@ def all_arrays(self):
         return self._all_arrays
 
     @all_arrays.setter
-    def all_arrays(self, value):
-        raise AttributeError(
-            "Direct modification of attribute 'all_arrays' is not allowed."
-            "Please use 'insert_scattering_quantity' to modify `all_arrays`."
-        )
+    def all_arrays(self, _):
+        raise AttributeError(_setter_wmsg("all_arrays"))
+
+    @property
+    def xtype(self):
+        return self._xtype
+
+    @xtype.setter
+    def xtype(self, _):
+        raise AttributeError(_setter_wmsg("xtype"))
```

Comment:
@sbillinge Like Would you also like to prevent the user from directly modifying `xtype`?